registry/storage/driver/azure/azure.go

```diff
@@ -489,8 +468,9 @@ func (w *writer) Cancel() error {
 		return fmt.Errorf("already committed")
 	}
 	w.cancelled = true
-	blobRef := w.driver.client.GetContainerReference(w.driver.container).GetBlobReference(w.path)
-	return blobRef.Delete(nil)
+	blobRef := w.driver.client.NewBlobClient(w.path)
+	_, err := blobRef.Delete(w.ctx, nil)
```

Comment:
If `w.ctx` is done before `Cancel()` is called, as is likely to be the case if the upload is canceled, the blob won't get deleted. Strangely enough, the consumer of `FileWriter` instances has a context available to pass in.
https://github.com/distribution/distribution/blob/0c958010ace2e0c2a87f1bf9915b7c74157dfb62/registry/storage/blobwriter.go#L95-L99
If you don't feel comfortable changing the `FileWriter` interface so that `Cancel` takes a context argument, I would feel much safer if the context passed into `newWriter` was not used here.
```suggestion
	_, err := blobRef.Delete(context.TODO(), nil)
```

Reply:
thanks, I missed that! I've changed the FileWriter interface to support a context argument to Cancel.

Reply:
I've completely missed this conversation, hence asking a related question on my review. Sorry about this, friends!